ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/RequiredPluginsInitializer.java

```diff
@@ -36,6 +37,22 @@ public class RequiredPluginsInitializer extends ClasspathContainerInitializer {
 
 	@Override
 	public void initialize(IPath containerPath, IJavaProject javaProject) throws CoreException {
+		if (Job.getJobManager().isSuspended()) {
+			// if the jobmanager is currently suspended we can't use the
+			// schedule/join pattern here, instead we must retry the requested
+			// action once jobs are enabled again, this will the possibly
+			// trigger a rebuild if required or notify other listeners.
+			Job job = Job.create(PDECoreMessages.PluginModelManager_InitializingPluginModels, m -> {
```

Comment:
I just tested this with my colleagues workspace and saw it creates 1 job per Project =>1 Worker Thread per Project. It works for me, but it looks dangerous many threads and each thread still (after https://github.com/eclipse-platform/eclipse.platform/pull/1720) burns significant time on synchronization. Could it be somehow refactored to not spam so many jobs, probably a single Job initializing all projects?
![image](https://github.com/user-attachments/assets/e6c1b6a7-ff86-4f6f-b9d5-d96e0ea378b6)